fuel/transformers/__init__.py

```diff
@@ -441,41 +547,38 @@ def sources(self):
                 sources.append(source + '_mask')
         return tuple(sources)
 
-    def get_data_from_batch(self, request=None):
-        if request is not None:
-            raise ValueError
-        data = list(next(self.child_epoch_iterator))
-        data_with_masks = []
-        for i, (source, source_data) in enumerate(
-                zip(self.data_stream.sources, data)):
+    def transform_batch(self, batch):
+        batch_with_masks = []
+        for i, (source, source_batch) in enumerate(
+                zip(self.data_stream.sources, batch)):
             if source not in self.mask_sources:
-                data_with_masks.append(source_data)
+                batch_with_masks.append(source_batch)
                 continue
 
-            shapes = [numpy.asarray(sample).shape for sample in source_data]
+            shapes = [numpy.asarray(sample).shape for sample in source_batch]
             lengths = [shape[0] for shape in shapes]
             max_sequence_length = max(lengths)
             rest_shape = shapes[0][1:]
             if not all([shape[1:] == rest_shape for shape in shapes]):
                 raise ValueError("All dimensions except length must be equal")
-            dtype = numpy.asarray(source_data[0]).dtype
+            dtype = numpy.asarray(source_batch[0]).dtype
 
-            padded_data = numpy.zeros(
-                (len(source_data), max_sequence_length) + rest_shape,
+            padded_batch = numpy.zeros(
+                (len(source_batch), max_sequence_length) + rest_shape,
                 dtype=dtype)
-            for i, sample in enumerate(source_data):
-                padded_data[i, :len(sample)] = sample
-            data_with_masks.append(padded_data)
+            for i, sample in enumerate(source_batch):
+                padded_batch[i, :len(sample)] = sample
+            batch_with_masks.append(padded_batch)
 
-            mask = numpy.zeros((len(source_data), max_sequence_length),
+            mask = numpy.zeros((len(source_batch), max_sequence_length),
                                self.mask_dtype)
             for i, sequence_length in enumerate(lengths):
                 mask[i, :sequence_length] = 1
-            data_with_masks.append(mask)
-        return tuple(data_with_masks)
+            batch_with_masks.append(mask)
+        return tuple(batch_with_masks)
 
 
-class Merge(Transformer):
+class Merge(AbstractDataStream):
```

Comment:
The `close`, `reset` and `next_epoch` methods defined in `Transformer` assume that the transformer wraps only one data stream, so I would need to override those. Plus, `Merge` already overrides `get_epoch_iterator`, which pretty much removes all the benefits from inheriting from `Transformer`. The only advantage of doing so would be to keep an *is a* relationship with `Transformer`, and even then, it's not 100% clear that merging data streams should be counted as a transformation.

What do you think?